components/canvas/webgl_thread.rs

```diff
@@ -1703,3 +1711,147 @@ fn map_dot_separated<F: Fn(&str, &mut String)>(s: &str, f: F) -> String {
     }
     mapped
 }
+
+fn profiler_category(webgl_command: &WebGLCommand) -> ProfilerCategory {
```

Comment:
This could be generated with a macro wrapping this function and the definition of `WebGLCommand`.